### PR TITLE
Fixes issues/141

### PR DESCRIPTION
**Pull request by Liath**
wrapped JSON parse in a try-except

### CODE REVIEW

File: yas3fs/__init__.py [hunk 1]
```diff
@@ -1211,7 +1211,13 @@ def delete_cache(self, path):
 
     def process_message(self, messages):
         logger.debug("process_message '%s'" % (messages))
-        c = json.loads(messages)
+        # Discard malformed JSON https://github.com/danilop/yas3fs/issues/141
+        try:
+            c = json.loads(messages)
+        except Exception as e:
+            logger.debug("process_message discarding malformed message")
```
Comment (dacut) [inline yas3fs/__init__.py]: Should this be more severe? (Warning, maybe even error?)
Also, is Exception too broad? (Probably harmless here, though.)

Comment (dacut) [inline yas3fs/__init__.py]: I like @bitsofinfo 's recommendation about logging the exception message. Maybe log the actual message to debug, too?

**Comment by Liath** [comment]
Something like this maybe?
```
logger.debug("parsing error: %s" % (e))
logger.warn("process_message discarding malformed message")
```

File: yas3fs/__init__.py (hunk 1 above)
Comment (Liath) [inline yas3fs/__init__.py]: Idk enough about python for the second question. Recommendations?

Comment (dacut) [inline yas3fs/__init__.py]: Leave Exception as-is, on second thought. (`json.loads` only raises `ValueError`, but there's no guarantee that a custom decoder won't raise, say, `AttributeError` or somesuch, and we probably want to be resilient for that case, too.)

Change `logger.debug()` to:
`logger.debug("process_message: malformed message: %s", messages)`

**Comment by Liath** [comment]
Should I merge myself? I'm used to my companies rules against merging your own changes lol

**Comment by dacut** [comment]
My hunch is as long as you get the thumbs up, you’re good to go on the merge.

That said, I went ahead and merged it so my neck is officially on the line, too. :-)

**Comment by Liath** [comment]
Thank you. I just felt like I might be treading too heavily, but you make a compelling point about sharing blame lol

**Pull request merged**